packages/react/src/modules/slack/components/ConnectedSlackChannelPicker/ConnectedSlackChannelPicker.tsx

```diff
@@ -0,0 +1,216 @@
+import Knock, { User } from "@knocklabs/client";
+import { useMemo, useState } from "react";
+import { ContainerObject, useSlackChannels, Channel } from "../..";
+
+import "./styles.css";
+
+type Props = {
+  user: User;
+  tenant: string;
+  connectionsObject: ContainerObject;
+  host?: string;
+  userToken: string;
+  knockSlackChannelId: string;
+  knockClientId: string;
+};
+
+export const ConnectedSlackChannelPicker: React.FC<Props> = ({
+  user,
+  tenant,
+  connectionsObject,
+  knockClientId,
+  host = "api.knock.app",
+  userToken,
+  knockSlackChannelId,
+}) => {
+  const knockClient = useMemo(() => {
+    const knock = new Knock(knockClientId, {
+      host: host,
+    });
+    knock.authenticate(user.id, userToken);
+
+    return knock;
+  }, [host, knockClientId, user.id, userToken]);
+
+  const [toggleForRefetch, setToggleForRefetch] = useState(false);
+  const [isSettingChannels, setIsSettingChannels] = useState(false);
+  const [showChannelPicker, setShowChannelPicker] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const { data, isLoading: isLoadingSlackChannels } = useSlackChannels(
+    tenant,
+    connectionsObject,
+    host,
+    knockClientId,
+    knockSlackChannelId,
+    userToken,
+    user,
+    toggleForRefetch,
+    setHasError,
+    hasError,
+  );
+
+  const isLoading = isSettingChannels || isLoadingSlackChannels;
+
+  const { channels } = data;
+
+  const handleOptionClick = async (channelId: string) => {
+    setIsSettingChannels(true);
+    const channelToUpdate = channels.find(
+      (channel: Channel) => channel.id === channelId,
+    );
+    if (channelToUpdate) {
+      const updatedChannels = channels.map((channel: Channel) =>
```

Comment:
Minor comment: I think we can remove `updatedChannels` and combine the two maps (or chain the filter and 2 maps)